src/test/java/org/elasticsearch/index/query/BaseTermQueryTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanTermQuery;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+@Ignore
+public abstract class BaseTermQueryTest<QB extends BaseQueryBuilder & Streamable> extends BaseQueryTestCase<QB> {
+    
+    public Object createRandomValueObject() {
+        Object value = null;
+        switch (randomIntBetween(0, 3)) {
+        case 0:
+            value = randomBoolean();
+            break;
+        case 1:
+            if (randomInt(10) > 0) {
+                value = randomAsciiOfLength(8);
+            } else {
+                // generate unicode string in 10% of cases
+                value = randomUnicodeOfLength(10);
+            }
+            break;
+        case 2:
+            value = randomInt(10000);
+            break;
+        case 3:
+            value = randomDouble();
+            break;
+        }
+        return value;
+    }
+
+    @Test
+    public void testValidate() throws QueryParsingException, IOException {
+        TermQueryBuilder queryBuilder = new TermQueryBuilder("all", "good");
```

Comment:
I find it a bit confusing to use the concrete TermQueryBuilder.validate() test here in the abstract test case, although I understand that the tested code lives in the base class. This way it's run for both test suites. Not a big problem, but maybe it would be okay to just have this in one of the tests.